tests/integration_tests/performance/test_network_tcp_throughput.py

```diff
@@ -353,6 +356,12 @@ def test_network_throughput(bin_cloner_path):
         'name': 'network_tcp_throughput'
     }
 
+    if gather_performance_data:
+        test_context.custom['results_path'] = os.path.join(
```

Comment:
`PERF_TEST_RESULTS_DIR` could be defined as a `pathlib.Path` object.

If so, we can avoid calling `os.path.join` and just define:

```python
test_context.custom['results_path']  = PERF_TEST_RESULTS_DIR / 'test_network_tcp_throughput.out'

Reply:
after the latest changes, this no longer applies here. Changes the `TEST_RESULTS_DIR` to be a `pathlib.Path` object